### DISCUSSION TITLE
Access uhabits data from another app

### DISCUSSION THREAD

**Discussion by SkinyMonkey** [iSoron/uhabits#1190] [Feature Requests]
I was looking into accessing uhabits' data from companion app to add some gamification content without tainting the app itself.

Of what I was reading (I'm not an android dev) a ContentProvider should be added to uhabits to allow the data access from another app.

What do you think about allowing another app to access the data?
And Is it possible to add this feature?

**Comment by hiqua** [comment]
What format should it have? We can already export to .csv, why is that not sufficient / what other options do you think should exist?

**Comment by SkinyMonkey** [comment]
Exporting is nice but I don't want to do it by hand each time I want to update the UI of another app.

**Comment by VA-MMXXII** [comment]
What about adding an auto-backup or auto-export feature to the uhabit/Habits Loop app that regularly updates the exported data in a known directory where the other app could access it?

**Reply by mshuflin** [reply]
What about an auto export of the csv file?

**Reply by VA-MMXXII** [reply]
That'd be great. The open source Android Tasks app (tasks.org) has an automatic daily CSV export feature, so I know this can be done. It'd be great!

**Reply by SkinyMonkey** [reply]
I mentioned ContentProvider because exporting the data to a file to make it accessible to another app is discouraged as mentioned in the following documentation points:

https://developer.android.com/reference/android/content/Context#MODE_MULTI_PROCESS
https://developer.android.com/reference/android/content/Context#MODE_WORLD_READABLE

But yeah, an auto export would work.

**Comment by SkinyMonkey** [comment]
As far as I understand uhabit don't export to the local filesystem.
I'm not even sure that android allow this, it could make sense that app stay in isolated filesystem spaces.
But maybe there's a shared space that all app can access?

**Comment by SkinyMonkey** [comment]
Maybe @iSoron has an advice on this? :)
